src/components/NewCommentForm/NewCommentForm.tsx

```diff
@@ -0,0 +1,105 @@
+import './NewCommentForm.scss';
+import React, { FormEvent, useState } from 'react';
+import { Comment } from '../../types';
+
+type Props = {
+  commentsList: Comment[] | undefined;
+  setCommentsList: (arg:Comment[]) => void;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+                                                  commentsList, setCommentsList,
+                                                }) => {
+  const [yourname, setYourname] = useState('');
+  const [youremail, setYouremail] = useState('');
+  const [yourcomment, setYourcomment] = useState('');
+  const [error, setError] = useState(false);
+
+  const addComment = (newComment: Comment) => {
+    if (commentsList) {
+      commentsList.unshift(newComment);
+      const copy = [...commentsList];
+
+      setCommentsList(copy);
+    }
+  };
+
+  const submitHandler = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (yourname && youremail && yourcomment && commentsList) {
```

Comment:
Also, when checking, you should check whether the value does not contain spaces (value.trim()), because even one space is considered as a non-empty value, i.e. true.